src/node/utils/Settings.js

```diff
@@ -438,6 +438,12 @@ exports.getEpVersion = function() {
   return require('ep_etherpad-lite/package.json').version;
 }
 
+// Provide favicon address by setting.faveicon and requestedRootAddress
+exports.getFaviconAddress = function (favicon, requestedRootAddress = "/") {
+  var urlRegex = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*).?$/g;
```

Comment:
This needs thorough testing

Reply:
Done.

Reply:
Link me to test coverage please.

Reply:
Well, this is the [test case](https://github.com/HMarzban/etherpad-lite/blob/fix/static_files/tests/backend/specs/pad.js) for `getFaviconAddress` function.